### DISCUSSION TITLE
Further clarify flows and the associated efficiency

### DISCUSSION THREAD

**Discussion by gnawin** [TulipaEnergy/TulipaEnergyModel.jl#432] [Ideas]
Maybe it is just me but I find it a bit unclear when try to understand flows and the associated efficiency. I think the definition can be made clearer, and the naming may need to be more explicit in some cases.

The description of $v^{flow}_{f,rp,k}$ is _flow between two assets_, and $p^{eff}_f$ is _flow efficiency_. Here is my question: in the balance constraints, when do we attach efficiency to the flows and when not?

My understanding is that one should consider flows as physical flows. If the energy loss/efficiency happens at the asset level, the efficiency should be attached in the corresponding asset balance constraint, as it is for consumer and hub balances (no efficiency attached) and storage balance (efficiency attached because of charging/discharging at the storage).

Then I had trouble understanding the conversion balance. I would think the conversion would only introduce losses to one flow, that is the outflow. But as it is, there is also efficiency for the inflow. What does this efficiency mean and what would be a user case for that? 

```math
\begin{aligned}
\sum_{f \in \mathcal{F}_{in}(a)} p^{eff}_f \cdot {v^{flow}_{f,rp,k}} = \sum_{f \in \mathcal{F}_{out}(a)} \frac{v^{flow}_{f,rp,k}}{p^{eff}_f}  
\end{aligned}
```

**Comment by gnawin** [comment]
This has been discussed with @datejada @g-moralesespana. For the conversion balance, we keep efficiencies for both inflows and outflows in order to give flexibility on how you want to conceptualize and model inflows and outflows, but to not double-counting, one of the efficiencies should be 1. Two action points are turned to issues #436 #437 .

**Reply by gnawin** [reply]
@suvayu this is what we discussed

**Comment by clizbe** [comment]
@gnawin Is there more we need to address here or can we close this discussion? ^_^

**Reply by gnawin** [reply]
Thanks 😄 , we can close this
